modules/kubernetes-addons/karpenter/main.tf

```
@@ -37,7 +37,7 @@ resource "aws_sqs_queue_policy" "this" {
 resource "aws_cloudwatch_event_rule" "this" {
   for_each = { for k, v in local.event_rules : k => v if var.enable_spot_termination }
 
-  name          = each.value.name
+  name          = join("-", [each.value.name, var.addon_context.eks_cluster_id])
```

I would prefer using the `name_prefix` over a fixed `name` here. Similar to https://github.com/terraform-aws-modules/terraform-aws-eks/blob/c7565e265e23cbc622041fc153193f490cfe948f/modules/karpenter/main.tf#L270.

```suggestion
  name_prefix   =  each.value.name
```